Added MHJ support

EricMarcil opened the pull request:
Added support for the MHJ robot (FS100 controller)

gavanderhoorn commented on motoman_mhj_support/urdf/mhj_macro.xacro:
```diff
@@ -0,0 +1,165 @@
+        <child link="${prefix}link_3_u"/>
+        <origin xyz="0 0 0.275" rpy="0 0 0" />
+        <axis xyz="0 -1 0" />
+        <limit lower="${radians(-45)}" upper="${radians(210)}" effort="65.9" velocity="${radians(200)}"/>
```
According to [this specsheet](https://yaskawaindia.in/wp-content/uploads/2012/10/MOTOMAN-MHJ.pdf) (and [the specs on the Motoman site](https://www.motoman.com/en-us/products/robots/industrial/assembly-handling/mh-series/mhjf)), the limits for `U` are: `[-290; 105]`.

Is the specsheet incorrect?

EricMarcil replied:
There are a few spec sheet where they give the range of motion of the U-axis relative to the ground and it is the compound angle of the L and U range.  So it is not incorrect and looks good from a marketing perspective but not useful for us.

EricMarcil replied:
You are right about the color, that was an over sight.  I'll get that fixed.

gavanderhoorn commented:
@EricMarcil: could you please rebase your PR on-top of `kinetic-devel`?

I've updated the CI configuration to also include a Melodic build, and it would be good to have your package tested against that ROS version as well.

EricMarcil commented:
@gavanderhoorn :
>>could you please rebase your PR on-top of kinetic-devel?

I'm not sure what you mean.  As far as I can tell, I'm on the kinetic-devel.

>> I've updated the CI configuration to also include a Melodic build, and it would be good to have your package tested against that ROS version as well.

I've just upgraded to Melodic and the packaged build against it without error.

gavanderhoorn commented:
Rebasing essentially replaces the starting point of your changes with what is currently the `HEAD` of `kinetic-devel`, instead of where you started your branch originally.

The following commands should do what is needed:

```bash
cd /to/where/you/have/motoman_experimental

# I'm assuming 'origin' points to 'ros-industrial/motoman_experimental' here.
# if you've given it a different name (perhaps 'upstream', use that)
git fetch origin

# just making sure the branch for this PR is active
git checkout MHJ-support

# actual rebase ontop of what the current state in this repository is (ie:
# 'ros-industrial/motoman_experimental').
# note: again using 'origin' here, adjust as necessary
git rebase origin/kinetic-devel

# if there were no errors (I'm not expecting any), push changes to remote
git push https://github.com/EricMarcil/motoman_experimental.git MHJ-support --force
```

Note: I use the full URL of your fork in the last command, as I don't know what label you used for it.

At this point your PR here will have been updated and we should get a build against Melodic as well.

EricMarcil commented:
@gavanderhoorn I've ran the command you indicated and got a "Everything up-to-date" at the end.  But, I don't really any changes anywhere indicating that something as actually changed.  Can I just do the "Merge pull request" at this point?

gavanderhoorn commented:
No, because it looks like the rebase didn't work.

Can you post the output of `git remote -v` here (after `cd /to/where/you/have/motoman_experimental`)?

And also here we'll want to do a squash-merge.

EricMarcil commented:
```
origin  https://github.com/EricMarcil/motoman_experimental.git (fetch)
origin  https://github.com/EricMarcil/motoman_experimental.git (push)
upstream        https://github.com/ros-industrial/motoman_experimental.git (fetch)
upstream        https://github.com/ros-industrial/motoman_experimental.git (push)
```

gavanderhoorn commented:
Ok, updated commands:

```bash
cd /to/where/you/have/motoman_experimental

git fetch upstream
git checkout MHJ-support

git rebase upstream/kinetic-devel
git push origin MHJ-support --force
```

EricMarcil commented:
OK, that looks better:

```
C:\Users\marcier\Documents\GitHub\motoman_experimental>git rebase upstream/kinetic-devel
First, rewinding head to replay your work on top of it...
Applying: Added MHJ support
Applying: mhj: indent two spaces per level.
Applying: mhj: uppercase series identifier.
Applying: Changed color to white

C:\Users\marcier\Documents\GitHub\motoman_experimental>git push origin MHJ-support --force
Enumerating objects: 58, done.
Counting objects: 100% (58/58), done.
Delta compression using up to 8 threads
Compressing objects: 100% (53/53), done.
Writing objects: 100% (57/57), 958.44 KiB | 12.61 MiB/s, done.
Total 57 (delta 17), reused 0 (delta 0)
remote: Resolving deltas: 100% (17/17), completed with 1 local object.
To https://github.com/EricMarcil/motoman_experimental.git
 + b6dcf7f...50199d7 MHJ-support -> MHJ-support (forced update)
```

So now, I just need for Travis to complete.  Once that is done. I can do the Merge?

gavanderhoorn commented:
As long as you do a squash-merge, feel free to press the button.

EricMarcil commented:
@gavanderhoorn Thanks for the assistance.

gavanderhoorn commented:
@EricMarcil: you could also remove this branch. It has been squash-merged as well.

EricMarcil commented:
@gavanderhoorn : Done
